src/transactions/ParallelApplyUtils.cpp

```diff
@@ -310,21 +328,20 @@ flushResidualRoTTLBumps(SearchableSnapshotConstPtr liveSnapshot,
     }
 }
 
-// Construct a map of all the TTL keys associated with all soroban
-// (code-or-data) keys named in the footprint of the `txBundle`, along with a
-// boolean indicating whether they're RO TTL keys. When false, they're RW.
-UnorderedMap<LedgerKey, bool>
-buildRoTTLMap(TxBundle const& txBundle)
+// Construct a set of all the TTL keys associated with all RO soroban
+// (code-or-data) keys named in the footprint of the `txBundle`.
+UnorderedSet<LedgerKey>
+buildRoTTLSet(TxBundle const& txBundle)
 {
-    UnorderedMap<LedgerKey, bool> isReadOnlyTTLMap;
+    UnorderedSet<LedgerKey> isReadOnlyTTLSet;
     for (auto const& ro :
          txBundle.getTx()->sorobanResources().footprint.readOnly)
     {
         if (!isSorobanEntry(ro))
         {
             continue;
         }
-        isReadOnlyTTLMap.emplace(getTTLKey(ro), true);
+        isReadOnlyTTLSet.emplace(getTTLKey(ro));
     }
     for (auto const& rw :
```

Comment:
Th rw loop can be removed since we're only dealing with a single TX here and we're guarenteed keys don't appear in multiple footprints, right?